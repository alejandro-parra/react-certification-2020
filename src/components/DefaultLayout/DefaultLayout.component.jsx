import React from 'react';
import Header from '../Header/Header.component';

const DefaultLayout = ({ children }) => {
  return (
    <>
      <Header />
      {children}
    </>
  );
};

export default DefaultLayout;
