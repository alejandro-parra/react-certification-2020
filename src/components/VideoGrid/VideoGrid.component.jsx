import React from 'react';
import VideoCard from '../VideoCard/VideoCard.component';
import { EmptyMessage, GridContainer } from './VideoGrid.styled';

const VideoGrid = ({ videos, clickHandler }) => {
  return (
    <GridContainer>
      {videos.map((video) => {
        return (
          <VideoCard
            key={video.id}
            video={video}
            mode="grid"
            clickHandler={clickHandler}
          />
        );
      })}
      {videos.length === 0 ? (
        <EmptyMessage>Actualmente no hay videos favoritos en tu cuenta.</EmptyMessage>
      ) : null}
    </GridContainer>
  );
};

export default VideoGrid;
