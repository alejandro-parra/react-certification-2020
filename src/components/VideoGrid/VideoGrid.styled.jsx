import styled from 'styled-components';

export const GridContainer = styled.div`
  width: 98%;
  min-height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: flex-start;
  align-content: flex-start;
  flex-flow: wrap;
  margin-top: 7rem;
`;

export const EmptyMessage = styled.p`
  padding: 3rem;
  width: 100%;
  min-height: 100vh;
  text-align: center;
`;
