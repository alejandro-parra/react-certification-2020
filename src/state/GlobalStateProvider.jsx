import React, { createContext, useContext, useReducer } from 'react';
import { AUTH_USERINFO_KEY, AUTH_FAVORITES_KEY } from '../utils/constants';
import { storage } from '../utils/storage';
import reducer from './GlobalStateReducer';

const initialState = {
  theme: 'light',
  userInfo: storage.get(AUTH_USERINFO_KEY)
    ? {
        ...storage.get(AUTH_USERINFO_KEY),
        favorites:
          storage.get(
            `${AUTH_FAVORITES_KEY}${storage.get(AUTH_USERINFO_KEY).googleId}`
          ) || [],
      }
    : null,
};

const GlobalContext = createContext({
  theme: '',
  userInfo: null,
});

function useGlobalState() {
  const context = useContext(GlobalContext);
  if (!context) {
    throw new Error(`Can't use "useGlobalState" without a GlobalStateProvider!`);
  }
  return context;
}

function GlobalStateProvider({ children }) {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <GlobalContext.Provider value={{ state, dispatch }}>
      {children}
    </GlobalContext.Provider>
  );
}

export { useGlobalState };

export default GlobalStateProvider;
