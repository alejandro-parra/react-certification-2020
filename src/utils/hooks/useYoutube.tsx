import React, { useState, useEffect } from 'react';
import axios from 'axios';
import apiKey from './../../data/apiKey';
import YoutubeSearchlistResponse from '../../model/YoutubeModels';
import VideoCardModel from '../../model/VideoCardModel';
import SearchType from '../../model/SearchType';

const fullURL = (query: string, searchType: SearchType) => {
  if (searchType === SearchType.searchTerm) {
    return `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=24&q=${query}&type=video&key=${apiKey}`;
  } else {
    return `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=12&relatedToVideoId=${query}&type=video&key=${apiKey}`;
  }
};

export const createVideoCardModel = (responseData: YoutubeSearchlistResponse) => {
  return responseData.items
    .filter(({ id }) => {
      return id.kind === 'youtube#video';
    })
    .map(({ etag, snippet }) => {
      const card: VideoCardModel = {
        id: etag,
        title: snippet.title,
        creationDate: snippet.publishTime,
        creator: snippet.channelTitle,
        thumbImage:
          snippet.thumbnails.high.url ??
          snippet.thumbnails.medium.url ??
          snippet.thumbnails.default.url,
      };
      return card;
    });
};

export const useYoutubeListFetcher = (query: string, searchType: SearchType) => {
  const [videoList, setVideoList] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true);
      try {
        let response = await axios.get(fullURL(query, searchType));
        let data = response.data as YoutubeSearchlistResponse;
        setVideoList(createVideoCardModel(data));
        setLoading(false);
      } catch (err) {
        setLoading(false);
        console.log(`Error while fetching data: ${err}`);
      }
    };
    fetchData();
  }, [query]);

  return { videoList, loading };
};
