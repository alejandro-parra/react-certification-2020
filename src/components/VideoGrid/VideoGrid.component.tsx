import React from 'react';
import VideoCard from '../VideoCard/VideoCard.component';
import { GridContainer } from './VideoGrid.styled';

const VideoGrid = ({ videos }) => {
  return (
    <GridContainer>
      {videos.map(({ id, title, creationDate, creator, thumbImage }) => {
        return (
          <VideoCard
            key={id}
            title={title}
            creationDate={creationDate}
            creator={creator}
            thumbImage={thumbImage}
          />
        );
      })}
    </GridContainer>
  );
};

export default VideoGrid;
