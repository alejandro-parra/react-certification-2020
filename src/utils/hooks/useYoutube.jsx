import { useState, useEffect } from 'react';
import axios from 'axios';
import mockVideos from '../../data/mock-videos.json';
import { useGlobalState } from '../../state/GlobalStateProvider';

const apiKey = process.env.REACT_APP_API_KEY;

export const fullURL = (query, searchType) => {
  if (searchType === 'search') {
    return `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=12&q=${query}&type=video&key=${apiKey}`;
  }
  return `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=12&relatedToVideoId=${query}&type=video&key=${apiKey}`;
};

export const createVideoCardModel = (responseData, favorites) => {
  const favList = favorites || [];
  return responseData.items
    .filter(({ id }) => {
      return id.kind === 'youtube#video';
    })
    .filter((video) => video.snippet)
    .map(({ id, snippet }) => {
      return {
        id: id.videoId,
        title: snippet.title,
        creationDate: snippet.publishTime,
        creator: snippet.channelTitle,
        thumbImage:
          snippet.thumbnails.high.url ??
          snippet.thumbnails.medium.url ??
          snippet.thumbnails.default.url,
        description: snippet.description,
        favorited: favList.filter((fav) => fav.id === id.videoId).length === 1,
      };
    });
};

const updateFavorites = (videoList, favorites) => {
  return videoList.map((video) => {
    return {
      ...video,
      favorited: favorites.filter((fav) => fav.id === video.id).length === 1,
    };
  });
};

export const useYoutubeListFetcher = (query, searchType) => {
  const { state } = useGlobalState();
  const [videoList, setVideoList] = useState([]);
  const [loading, setLoading] = useState(false);

  const loadNewVideos = async (newQuery, newType) => {
    const favorites = state.userInfo ? state.userInfo.favorites || [] : [];
    setLoading(true);
    if (newType === 'favorites') {
      setVideoList(favorites);
      setLoading(false);
    } else {
      try {
        const response = await axios.get(fullURL(newQuery, newType));
        const data = { ...response.data };
        setVideoList(createVideoCardModel(data, favorites));
        setLoading(false);
      } catch (err) {
        const data = { ...mockVideos };
        setVideoList(createVideoCardModel(data, favorites));
        setLoading(false);
      }
    }
  };

  useEffect(() => {
    loadNewVideos(query, searchType);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    if (state.userInfo) {
      setVideoList(updateFavorites(videoList, state.userInfo.favorites || []));
      if (searchType === 'favorites') {
        setVideoList(state.userInfo.favorites);
      }
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [state]);
  return { videoList, loadNewVideos, loading };
};
