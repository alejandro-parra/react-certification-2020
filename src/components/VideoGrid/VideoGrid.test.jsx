import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import VideoGrid from './VideoGrid.component';
import mockVideos from '../../data/mock-videos.json';
import DefaultLayout from '../DefaultLayout/DefaultLayout.component';
import { createVideoCardModel } from '../../utils/hooks/useYoutube';
import AuthProvider from '../../providers/Auth';
import AppContainer from '../App/App.styled';
import GlobalStateProvider from '../../state/GlobalStateProvider';
import theme from '../../utils/theme';

describe('Video Grid', () => {
  it('renders properly with mock videos', () => {
    const videoList = createVideoCardModel({ ...mockVideos }, []);
    const result = create(
      <GlobalStateProvider>
        <ThemeProvider theme={theme.light}>
          <AppContainer>
            <AuthProvider>
              <DefaultLayout>
                <VideoGrid videos={videoList} clickHandler={() => {}} />
              </DefaultLayout>
            </AuthProvider>
          </AppContainer>
        </ThemeProvider>
      </GlobalStateProvider>
    ).toJSON();
    expect(result).toMatchSnapshot();
  });
});
