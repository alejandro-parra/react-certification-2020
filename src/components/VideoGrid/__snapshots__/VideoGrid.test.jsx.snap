// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Grid renders properly with mock videos 1`] = `
<div
  className="sc-dmlrTW kinGiG"
>
  <div
    className="sc-fFubgz bAGdYy"
  >
    <div
      className="sc-idOhPF sc-dIUggk iYJWlF jfHVNa"
    >
      <img
        alt="Logo"
        className="sc-bkzZxe jTLLhH"
        onClick={[Function]}
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    <div
      className="sc-crrsfI hpTrWk"
    >
      <input
        className="sc-dQppl kZWgzQ"
        onChange={[Function]}
        placeholder="Busca un video"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root sc-bqyKva dDZCed"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      className="sc-idOhPF sc-hHftDr iYJWlF rxi"
    >
      <div
        className="MuiFormGroup-root"
      >
        <label
          className="MuiFormControlLabel-root"
          role="switch"
        >
          <span
            className="MuiSwitch-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-4 MuiSwitch-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="MuiSwitch-thumb"
                />
              </span>
            </span>
            <span
              className="MuiSwitch-track"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Claro
          </span>
        </label>
      </div>
      <div
        className="sc-kstrdz jOePYU"
        onClick={[Function]}
      >
        <div
          className="sc-hBEYos cXOUbc"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="sc-kEjbxe elJRDT"
  >
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Video Tour | Welcome to Wizeline Guadalajara"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/nmXMgqjQzls/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Video Tour | Welcome to Wizeline Guadalajara
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 1 año
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline Guadalajara | Bringing Silicon Valley to Mexico"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/HYyRZiwBWc8/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline Guadalajara | Bringing Silicon Valley to Mexico
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 2 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline hace sentir a empleados como en casa"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/Po3VwR_NNGk/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline hace sentir a empleados como en casa
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          El Economista TV
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 2 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="We Are Wizeline"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/7PtYNO6g7eI/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          We Are Wizeline
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 2 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline Thrives in Mexico City"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/YuW0CE_8i1I/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline Thrives in Mexico City
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 2 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/CHzlSGRvWPs/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Noticieros Televisa
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 4 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="A Day in the Life of an Engineering Manager at Wizeline"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/cjO2fJy8asM/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          A Day in the Life of an Engineering Manager at Wizeline
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 2 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline Technical Writing Academy | Featuring Eduardo Ocejo"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/zClI9OjgKXM/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline Technical Writing Academy | Featuring Eduardo Ocejo
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 1 año
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Silicon Valley en México"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/8bz9R61oY5o/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Silicon Valley en México
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Azteca Noticias
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 1 año
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Why Wizeline? featuring Juan Pablo Villa in Mexico City"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/7dJFraOqcoQ/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Why Wizeline? featuring Juan Pablo Villa in Mexico City
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 1 año
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline&#39;s 2020 Year in Review"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/w-Qwc_XJrWc/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline&#39;s 2020 Year in Review
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 3 meses
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Why Wizeline? featuring Hugo Lopez in Mexico City"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/rjir_cHTl5w/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Why Wizeline? featuring Hugo Lopez in Mexico City
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 2 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Why Wizeline? (We&#39;re Hiring in Mexico!)"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/DcFK1x3NHGY/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Why Wizeline? (We&#39;re Hiring in Mexico!)
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 4 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Why Wizeline? featuring Oswaldo Herrera in Mexico City"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/3BzYWAqZgFw/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Why Wizeline? featuring Oswaldo Herrera in Mexico City
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 1 año
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Caso de Éxito AWS: Wizeline [Spanish]"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/3KVFmT-Tp2w/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Caso de Éxito AWS: Wizeline [Spanish]
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Amazon Web Services
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 2 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline&#39;s 2019 Year in Review"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/aKuPmY2m1Ro/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline&#39;s 2019 Year in Review
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 1 año
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="[1 of 2] Wizeline CEO shares career lessons from Google"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/24sTHUyWhRM/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          [1 of 2] Wizeline CEO shares career lessons from Google
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 4 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline Data Engineering featuring Tania Reyes"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/IxGc1gSqB3A/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline Data Engineering featuring Tania Reyes
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 2 meses
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Wizeline Querétaro | Mexico&#39;s New Knowledge Economy (We&#39;re hiring!)"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/NP1gAnbeNno/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Wizeline Querétaro | Mexico&#39;s New Knowledge Economy (We&#39;re hiring!)
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 1 año
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Miriam Godinez | Women in Leadership at Wizeline"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/F6Krwu6lUc8/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Miriam Godinez | Women in Leadership at Wizeline
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 6 meses
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Welcome Back to Wizeline Vietnam | Extended Version"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/RFq7gfvhtCk/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Welcome Back to Wizeline Vietnam | Extended Version
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 11 meses
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="How does Wizeline work?"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/E1Vq_A3WKK8/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          How does Wizeline work?
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 3 años
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="12 Wishes from Wizeline | Happy Holidays 2019"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/ZmkslANDz0Q/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          12 Wishes from Wizeline | Happy Holidays 2019
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 1 año
        </p>
      </div>
    </div>
    <div
      className="sc-bdfBwQ sc-gsTCUz boSqpk hjlWjc"
      onClick={[Function]}
      role="button"
    >
      <div
        className="sc-hKgILt cyPSrD"
      >
        <img
          alt="Why Wizeline?"
          className="sc-eCssSg bucUSe"
          src="https://i.ytimg.com/vi/Nss3EmTDD3s/hqdefault.jpg"
        />
      </div>
      <div
        className="sc-jSgupP kQvcdl"
      >
        <p
          className="sc-gKsewC bQiBLQ"
        >
          Why Wizeline?
        </p>
        <p
          className="sc-iBPRYJ sc-fubCfw iINBNA eBNBg"
        >
          Wizeline
        </p>
        <p
          className="sc-iBPRYJ iINBNA"
        >
          Hace 3 años
        </p>
      </div>
    </div>
  </div>
</div>
`;
